dbms/programs/client/Client.cpp

```diff
@@ -796,14 +796,33 @@ class Client : public Poco::Util::Application
         written_progress_chars = 0;
         written_first_block = false;
 
-        connection->forceConnected();
+        {
+            /// Temporarily apply query settings to context.
```

Comment:
Why do we need to do apply all server-side settings on the client-side here?

Reply:
It doesn't seem to be trivial to distinguish at this point which setting are client-side and which are server-side. It's easier to just apply all the settings.